crates/ruff_linter/src/rules/pylint/rules/sys_exit_alias.rs

```diff
@@ -67,15 +67,42 @@ pub(crate) fn sys_exit_alias(checker: &Checker, func: &Expr) {
         SysExitAlias {
             name: builtin.to_string(),
         },
-        func.range(),
+        call.func.range(),
     );
+
+    let num_str = call
+        .arguments
+        .args
+        .first()
+        .and_then(|arg| arg.as_number_literal_expr())
+        .map(|lit| format!("{:?}", lit.value))
+        .or_else(|| {
+            call.arguments
+                .keywords
+                .first()
+                .filter(|kwr| kwr.value.is_number_literal_expr())
+                .map(|kwr| format!("{:?}", kwr.value.as_number_literal_expr().unwrap().value))
+        });
+
+    let code = if let Some(value) = num_str {
+        value
+            .strip_prefix("Int(")
+            .and_then(|s| s.strip_suffix(")"))
+            .and_then(|num_str| num_str.parse::<i32>().ok())
+            .unwrap()
+    } else {
+        // if it is not possible to retrieve the code, just a violation is raised
+        checker.report_diagnostic(diagnostic);
+        return;
+    };
```

Comment:
I think you can simplify this to:

```suggestion
    let arg = call.arguments.find_argument_value("code", 0);

    let code = if let Some(arg) = arg {
        &checker.source()[arg.range()]
    } else {
        // if it is not possible to retrieve the code, just a violation is raised
        checker.report_diagnostic(diagnostic);
        return;
    };
```

`Arguments::find_argument_value` is a really handy method for this kind of thing, and we can just slice the `checker.source` instead of doing the parsing stuff you have here. `find_argument_value` will already ensure the argument isn't starred (as in your nice `**dict` case). Additionally, this will allow us to accept variable arguments that are not starred. I'm surprised that none of existing test cases had something like this:

```python
if success:
    code = 0
else:
    code = 1

exit(code)
```

We should add a test case where we fix that because I think it's a reasonable use case.

Reply:
Even this is slightly overly-restrictive because it would exclude a valid starred expression like:

```python
import sys

code = [1]
sys.exit(*code)
```

but that's arguably an issue with `find_positional` itself, I think.